sample/signalr-server-selfhost/Connections/DemoHub.cs

```diff
@@ -0,0 +1,248 @@
+        {
+            await Task.Factory.StartNew(() =>
+            {
+                throw new Exception();
+            });
+        }
+
+        public async Task<int> GenericTaskWithException()
+        {
+            return await Task<int>.Factory.StartNew(() =>
+            {
+                throw new Exception();
+            });
+        }
+
+        public void SynchronousException()
+        {
+            throw new Exception();
+        }
+
+        public void HubException()
+        {
+            throw new HubException("message", "errorData");
+        }
+
+        public void HubExceptionWithoutErrorData()
+        {
+            throw new HubException("message");
+        }
+
+        public Task CancelledTask()
+        {
+            var tcs = new TaskCompletionSource<object>();
+            tcs.SetCanceled();
+            return tcs.Task;
+        }
+
+        public Task<int> CancelledGenericTask()
+        {
+            var tcs = new TaskCompletionSource<int>();
+            return Task.Factory.StartNew(() =>
+            {
+                tcs.SetCanceled();
+                return tcs.Task;
+            }).Unwrap();
+        }
+
+        public Task NeverEndingTask()
+        {
+            return _neverEndingTcs.Task;
+        }
+
+        public void SimpleArray(int[] nums)
+        {
+            foreach (var n in nums)
+            {
+            }
+        }
+
+        public string ReadStateValue()
+        {
+            return Clients.Caller.name;
+        }
+
+        public string SetStateValue(string value)
+        {
+            Clients.Caller.Company = value;
+
+            return Clients.Caller.Company;
+        }
+
+        public object ReadAnyState()
+        {
+            Clients.Caller.state2 = Clients.Caller.state;
+            Clients.Caller.addy = Clients.Caller.state.Address;
+
+            string name = Clients.Caller.state["Name"];
+            string street = Clients.Caller.state["Address"]["Street"];
+
+            string dname = Clients.Caller.state.Name;
+            string dstreet = Clients.Caller.state.Address.Street;
+
+            if (!name.Equals(dname))
+            {
+                throw new InvalidOperationException("Fail");
+            }
+
+            if (!street.Equals(dstreet))
+            {
+                throw new InvalidOperationException("Fail");
+            }
+
+            return Clients.Caller.state;
+        }
+
+        public void ComplexArray(Person[] people)
+        {
+
+        }
+
+        public Person ComplexType(Person p)
+        {
+            Clients.Caller.person = p;
+            return p;
+        }
+
+        public int PassingDynamicComplex(dynamic p)
+        {
+            return p.Age;
+        }
+
+        public void MultipleCalls()
+        {
+            for (int i = 0; i < 10; i++)
+            {
+                Clients.Caller.index = i + 1;
+                Clients.Caller.invoke(i);
+                Thread.Sleep(1000);
```

Comment:
Why do we need Sleep like this (i.e. what scenarios are being tested) - it will make the tests run longer.